src/main/java/prefs/Preferences.java

```diff
@@ -15,83 +17,134 @@
-    public static final String TEST_CONFIG_FILE = "test.json";
-
-    private final ConfigFileHandler fileHandler;
+    public static final String SESSION_CONFIG_FILENAME = "global.json";
+    public static final String USER_CONFIG_FILENAME = "user.json";
 
-    public GlobalConfig global;
+    private final String sessionConfigFileName;
+    private final String userConfigFileName;
 
-    private Preferences(String configFileName, boolean createUnconditionally) {
-        this.fileHandler = new ConfigFileHandler(DIRECTORY, configFileName);
+    private final SessionConfig sessionConfig;
+    private final UserConfig userConfig;
 
-        if (createUnconditionally) {
-            initGlobalConfig();
-        } else {
-            loadGlobalConfig();
-        }
+    /**
+     * Initialises a Preferences instance that contain the configurations in the config files specified
+     * @param sessionConfigFileName The file name of the session config file which stores session config values.
+     * @param userConfigFileName The file name of the user config file which stores user config values.
+     * @return The initialised Preferences instance
+     */
+    public static Preferences load(String sessionConfigFileName, String userConfigFileName) {
+        return new Preferences(sessionConfigFileName, userConfigFileName, false);
     }
 
     /**
-     * Initialises a Preferences instance which creates its config file, or loads
-     * from it if it already exists.
+     * Initialises a Preferences instance that contain a clean slate of configuration values in the config files
+     * specified
+     * @param sessionConfigFileName The file name of the session config file which stores session config values.
+     * @param userConfigFileName The file name of the user config file which stores user config values.
+     * @return The initialised Preferences instance
      */
-    public static Preferences load(String configFileName) {
-        return new Preferences(configFileName, false);
+    public static Preferences create(String sessionConfigFileName, String userConfigFileName) {
+        return new Preferences(sessionConfigFileName, userConfigFileName, true);
     }
 
     /**
-     * Initialises a Preferences instance which always creates its config file.
-     * This will overwrite it with a default configuration if it already exists.
+     * @param sessionConfigFileName The name of the session config file
+     * @param ignoreExisting True if existing config file is to be ignored.
      */
-    public static Preferences create(String configFileName) {
-        return new Preferences(configFileName, true);
+    private Preferences(String sessionConfigFileName, String userConfigFileName, boolean ignoreExisting) {
+        this.sessionConfigFileName = sessionConfigFileName;
+        this.userConfigFileName = userConfigFileName;
+
+        this.sessionConfig = (SessionConfig) loadConfig(sessionConfigFileName, SessionConfig.class, ignoreExisting);
```

Comment:
I think you have to propagate the type constraints to `JsonHelper` as well, i.e. it should also return a `<T extends Config>`.